odoo/modules/registry.py

```diff
@@ -472,23 +474,33 @@ def check_indexes(self, cr, model_names):
                    [tuple(row[0] for row in expected)])
         existing = {row[0] for row in cr.fetchall()}
 
-        if not self.has_trigram and any(row[3] == 'trigram' for row in expected):
+        # TODO: Move this into `_create_empty_database`
+        if not self.has_trigram and any(row[3] == 'trigram' for row in expected if row[0] not in existing):
             self.has_trigram = sql.install_pg_trgm(cr)
 
-        for indexname, tablename, columnname, index in expected:
+        for indexname, tablename, column_name, index, unaccent in expected:
             assert index in ('btree', 'btree_not_null', 'trigram', True, False, None)
             if index and indexname not in existing:
+                column_expression = f'"{column_name}"'
                 method = 'btree'
                 operator = ''
                 where = ''
                 if index == 'btree_not_null':
-                    where = f'"{columnname}" IS NOT NULL'
+                    where = f'{column_expression} IS NOT NULL'
                 elif index == 'trigram' and self.has_trigram:
                     method = 'gin'
                     operator = 'gin_trgm_ops'
+                    # add `unaccent` to the trigram index only because they are used mainly (TODO: only ?) for (i/=)like search
+                    # and unaccent is added only in these cases by the ORM.
+                    if unaccent and self.has_unaccent:
+                        if self.has_unaccent != 'i':
+                            # use `warnings.warn` to avoid to repeat the warning for each module
+                            warnings.warn("Unaccent is activated but the `unaccent` PostgreSQL function isn't immutable, cannot create appropriate index!", stacklevel=0)
+                        else:
+                            column_expression = get_unaccent_wrapper(cr)(column_expression)
```

Comment:
Shouldn't `btree` also get `unaccent` if available?

Maybe btree should have both `unaccent` and regular indexes, as unaccent is used or LIKE-type queries but not `=`?

Reply:
IMO, it is not worth it:

- The btree with `unaccent` will be only in one case: when we use the operators `=like` and `=ilike` (**we don't add unaccent for equality** and (not) (i)like operator will always have wildcard at the beginning then the btree won't be used) and when the right operand is not begins by a wildcard. -> These operators are not used a lot and the most of the time it targets small table (account.account for code by example).
- It will double the number of index for `varchar` column which are quite costly.
- Also it still possible to add manually if it is slow.